File: src/main/java/de/lulonaut/Bot/utils/GetGuild.java
package de.lulonaut.Bot.utils;

import de.lulonaut.wrapper.APIWrapper;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

public class GetGuild {

    public static String getGuild(String Username) {
        if (Conf.Endpoint.equals("slothpixel")) {
            JSONObject gResponse;
            try {
                gResponse = API.readJsonFromUrl("https://api.slothpixel.me/api/guilds/" + Username);
            } catch (IOException e) {
                return "";
            }
            if (gResponse != null && gResponse.has("guild")) {
                return "";
            }

            try {
                return gResponse != null ? gResponse.getString("name") : "";
            } catch (JSONException e) {
                return "";
            }
        } else if (Conf.Endpoint.equals("hypixel")) {
            try {
                de.lulonaut.wrapper.core.API API = APIWrapper.create(Conf.APIKey, true);
                String guild = API.getGuildByUsername(Username).getName();
                if (guild == null) {
                    return "";
                } else {
                    return guild;
                }
            } catch (Exception e) {
                return "";

            }
        } else {
            return "";
        }
    }
}

File: src/main/kotlin/de/lulonaut/bot/utils/GetGuild.kt
package de.lulonaut.bot.utils

import java.io.IOException
import java.lang.Exception
import de.lulonaut.wrapper.APIWrapper
import org.json.JSONException
import org.json.JSONObject

object GetGuild {
    fun getGuild(Username: String): String {
        return when (Conf.Endpoint) {
            "slothpixel" -> {
                val gResponse: JSONObject? = try {
                    API.readJsonFromUrl("https://api.slothpixel.me/api/guilds/$Username")
                } catch (e: IOException) {
                    return ""
                }
                if (gResponse != null && gResponse.has("guild")) {
                    return ""
                }
                try {
                    if (gResponse != null) gResponse.getString("name") else ""
                } catch (e: JSONException) {
                    ""
                }
            }
            "hypixel" -> {
                try {
                    val api = APIWrapper.create(Conf.APIKey, true)
                    val guild = api.getGuildByUsername(Username).name
                    guild ?: ""
                } catch (e: Exception) {
                    ""
                }
            }
            else -> {
                ""
            }
        }
    }
}